cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll

```diff
@@ -38,16 +46,6 @@ predicate memberMayBeVarSize(Class c, MemberVariable v) {
       // arithmetic is performed on the result
       so.getParent*() instanceof AddExpr
     )
-    or
-    exists(AddressOfExpr aoe |
-      // `&(c.v)` is taken
-      aoe.getAddressable() = v
-    )
-    or
-    exists(BuiltInOperationBuiltInOffsetOf oo |
-      // `offsetof(c, v)` using a builtin
-      oo.getAChild().(VariableAccess).getTarget() = v
-    )
```

Comment:
Why did these cases have to go? Won't this mean we consider fewer fields to be variable sized?

Reply:
The previous requirement was:
Field size less equal 1 and is is the last field in the structure and one of
* `sizeof` was used in an arithmetic expression (addition) 
* Address was taken
* `offsetof` is used on the field

This has been simplified to: 
Field size less equal 1 and it is the last field in the structure and
* if `sizeof` is taken then it is used in an arithmetic expression. 

So in cases where sizeof is used without addition we might consider fewer fields as variable sized.
My reasoning is that we should consider any and all fields with a size less equal to 1 (and its the last field in the structure) as variable size, unless we can argue otherwise.

Reply:
Alright, that makes sense. I like this simpler interpretation a lot more 👍